File: app/src/main/java/ua/hospes/rtm/data/sessions/operations/SetDriverOperation.java
package ua.hospes.rtm.data.sessions.operations;

import android.content.ContentValues;

import com.squareup.sqlbrite.BriteDatabase;

import ua.hospes.dbhelper.Operation;
import ua.hospes.dbhelper.builder.UpdateQuery;
import ua.hospes.dbhelper.builder.conditions.Condition;
import ua.hospes.rtm.core.db.tables.Sessions;

/**
 * @author Andrew Khloponin
 */
public class SetDriverOperation implements Operation<Integer> {
    private final int sessionId;
    private final long driverId;

    public SetDriverOperation(int sessionId, int driverId) {
        this.sessionId = sessionId;
        this.driverId = driverId;
    }

    @Override
    public Integer doOperation(BriteDatabase db) {
        UpdateQuery builder = new UpdateQuery(Sessions.name).where(Condition.eq(Sessions.ID, sessionId));

        ContentValues cv = new ContentValues();
        cv.put(Sessions.DRIVER_ID.name(), driverId);

        db.update(builder.getTable(), cv, builder.getWhereClause(), builder.getWhereArgs());

        return sessionId;
    }
}

File: app/src/main/java/ua/hospes/rtm/core/db/sessions/operations/SetDriverOperation.kt
package ua.hospes.rtm.core.db.sessions.operations

import android.content.ContentValues

import com.squareup.sqlbrite2.BriteDatabase

import ua.hospes.dbhelper.Operation
import ua.hospes.dbhelper.builder.UpdateQuery
import ua.hospes.dbhelper.builder.conditions.Condition
import ua.hospes.rtm.core.db.tables.Sessions

/**
 * @author Andrew Khloponin
 */
class SetDriverOperation(private val sessionId: Int, driverId: Int) : Operation<Int> {
    private val driverId: Long

    init {
        this.driverId = driverId.toLong()
    }

    override fun doOperation(db: BriteDatabase): Int? {
        val builder = UpdateQuery(Sessions.name).where(Condition.eq(Sessions.ID, sessionId))

        val cv = ContentValues()
        cv.put(Sessions.DRIVER_ID.name(), driverId)

        db.update(builder.table, cv, builder.whereClause, *builder.whereArgs)

        return sessionId
    }
}